editor/src/components/canvas/controls/select-mode/remix-scene-label.tsx

```diff
@@ -0,0 +1,283 @@
+      }
+    },
+    [dispatch, scale, canvasOffset, props.target, onMouseUp],
+  )
+
+  React.useEffect(() => {
+    return () => {
+      window.removeEventListener('mouseup', onMouseUp, true)
+    }
+  }, [onMouseUp])
+
+  const backgroundColor = isSelected ? colorTheme.aqua05solid.value : 'transparent'
+
+  if (frame != null && isFiniteRectangle(frame)) {
+    return (
+      <CanvasOffsetWrapper>
+        <FlexRow
+          onMouseOver={labelSelectable ? onMouseOver : NO_OP}
+          onMouseOut={labelSelectable ? onMouseLeave : NO_OP}
+          onMouseDown={labelSelectable ? onMouseDown : NO_OP}
+          onMouseMove={labelSelectable ? onMouseMove : NO_OP}
+          data-testid={RemixSceneLabelTestID(props.target)}
+          className='roleComponentName'
+          style={{
+            pointerEvents: labelSelectable ? 'initial' : 'none',
+            color: isIndexRoute ? colorTheme.aqua.value : colorTheme.textColor.value,
+            position: 'absolute',
+            left: frame.x,
+            bottom: -frame.y + offsetY,
+            width: frame.width,
+            paddingLeft: offsetX,
+            paddingTop: paddingY,
+            paddingBottom: paddingY,
+            fontFamily: 'Utopian-Inter',
+            fontSize: scaledFontSize,
+            lineHeight: `${scaledLineHeight}px`,
+            whiteSpace: 'nowrap',
+            overflow: 'hidden',
+            textOverflow: 'ellipsis',
+            borderRadius: borderRadius,
+            backgroundColor: backgroundColor,
+            gap: 12 / scale,
+          }}
+        >
+          <Tooltip title={'Back'}>
+            <span
+              data-testid={RemixSceneLabelButtonTestId(props.target, 'back')}
+              style={{ cursor: 'pointer', fontSize: 16 / scale }}
+              onMouseDown={back}
+            >
+              〱
+            </span>
+          </Tooltip>
+          <Tooltip title={'Forward'}>
+            <span
+              data-testid={RemixSceneLabelButtonTestId(props.target, 'forward')}
+              style={{ cursor: 'pointer', fontSize: 16 / scale, transform: 'scale(-1, 1)' }}
+              onMouseDown={forward}
+            >
+              〱
+            </span>
+          </Tooltip>
+          <Tooltip title={'Home'}>
+            <span
+              data-testid={RemixSceneLabelButtonTestId(props.target, 'home')}
+              style={{ cursor: 'pointer', fontSize: 16 / scale }}
+              onMouseDown={home}
+            >
+              ／
+            </span>
+          </Tooltip>
+          <div
+            data-testid={RemixSceneLabelPathTestId(props.target)}
+            style={{
+              backgroundColor: '#f2f3f4',
+              borderRadius: 10 / scale,
+              padding: `${4 / scale}px ${12 / scale}px`,
+              fontSize: 14 / scale,
+            }}
+          >
+            {label}
+          </div>
+        </FlexRow>
+      </CanvasOffsetWrapper>
+    )
+  } else {
+    return null
```

Comment:
Minor stylistic nitpick (feel free to ignore it), but I think a "happy path to the left" early return for the null case is more readable than this else at the end.

Reply:
refactored it in https://github.com/concrete-utopia/utopia/pull/4195/commits/82830b7462b903d14ae260bc804d14f3ae58515d